core/src/HIP/Kokkos_HIP_Instance.hpp

```diff
@@ -133,6 +135,7 @@ class HIPInternal {
   // Resizing of reduction related scratch spaces
   size_type *scratch_space(const std::size_t size);
   size_type *scratch_flags(const std::size_t size);
+  size_type *scratch_functor(const std::size_t size) const;
```

Comment:
Why pretend it is const if all other aren't marked as const and it forces you to make member data mutable?

Reply:
Now I see that is also how CUDA does it...

Reply:
The function needs to be `const` because it is called from a const object.